Examples/Modules/nuclear_fusion/analysis_proton_boron_fusion.py

```diff
@@ -0,0 +1,700 @@
+    ## Only 900 particles pairs per cell here because we ignore the 10% of protons that are at rest
+    expected_fusion_number = check_macroparticle_number(data,
+                                                        fusion_probability_target_value = 0.02,
+                                                        num_pair_per_cell = 900)
+    E_com = compute_E_com2(data)
+    check_alpha_yield(data, expected_fusion_number, E_com, proton_density = 1.e20,
+                                                           boron_density = 1.e26)
+    check_initial_energy2(data)
+
+def specific_check3(data):
+    check_isotropy(data, relative_tolerance = 7.e-2)
+    check_thermal_alpha_yield(data)
+
+def specific_check4(data):
+    ## In test 4, the boron initial density is so small that all borons should have fused within a
+    ## timestep dt. We thus assert that no boron remains at the end of the simulation.
+    assert(not boron_remains(data))
+
+def specific_check5(data):
+    ## Test 5 is similar to test 4, expect that the parameter fusion_probability_threshold is
+    ## increased to the point that we should severely underestimate the fusion yield. Consequently,
+    ## there should still be borons at the end of the test, which we verify here.
+    assert(boron_remains(data))
+
+def check_charge_conservation(rho_start, rho_end):
+    assert(np.all(is_close(rho_start, rho_end, rtol=2.e-11)))
+
+def main():
+    filename_end = sys.argv[1]
+    filename_start = filename_end[:-4] + '0000'
+    ds_end = yt.load(filename_end)
+    ds_start = yt.load(filename_start)
+    ad_end = ds_end.all_data()
+    ad_start = ds_start.all_data()
+    field_data_end = ds_end.covering_grid(level=0, left_edge=ds_end.domain_left_edge,
+                                          dims=ds_end.domain_dimensions)
+    field_data_start = ds_start.covering_grid(level=0, left_edge=ds_start.domain_left_edge,
+                                              dims=ds_start.domain_dimensions)
+
+    ntests = 5
+    for i in range(1, ntests+1):
+        proton_species = "proton"+str(i)
+        boron_species = "boron"+str(i)
+        alpha_species = "alpha"+str(i)
+        data = {}
+        add_species_to_dict(ad_start, data, proton_species, "proton", "start")
+        add_species_to_dict(ad_start, data, boron_species, "boron", "start")
+        add_species_to_dict(ad_end, data, proton_species, "proton", "end")
+        add_species_to_dict(ad_end, data, boron_species, "boron", "end")
+        add_species_to_dict(ad_end, data, alpha_species, "alpha", "end")
+
+        # General checks that are performed for all tests
+        generic_check(data)
+
+        # Checks that are specific to test number i
+        eval("specific_check"+str(i)+"(data)")
+
+    rho_start = field_data_start["rho"].to_ndarray()
+    rho_end = field_data_end["rho"].to_ndarray()
+    check_charge_conservation(rho_start, rho_end)
+
+    test_name = filename_end[:-9] # Could also be os.path.split(os.getcwd())[1]
```

Comment:
Let's use the general version, please see #2385 / #2683:
```suggestion
    test_name = os.path.split(os.getcwd())[1]
```